packages/frontend/src/utils/twoFactor/twoFactorBase.js

```diff
@@ -0,0 +1,214 @@
+        if (isLedgerEnabled) {
+            throw new WalletError('Ledger Hardware Wallet is enabled', 'initTwoFactor.ledgerEnabled');
+        }
+
+        // clear any previous requests in localStorage (for verifyTwoFactor)
+        this.setRequest({ requestId: -1 });
+        return await this.init2fa({ accountId, method });
+    }
+
+    async getMultisigRequest() {
+        const { requestId, accountId } = this.getRequest();
+        return {
+            ...await this.viewFunction(this.accountId, 'get_request', { request_id: requestId }),
+            request_id: requestId,
+            account_id: accountId,
+        };
+    }
+
+    async deployMultisig() {
+        const contractBytes = new Uint8Array(await (await fetch('/multisig.wasm')).arrayBuffer());
+        await super.deployMultisig(contractBytes);
+        this.has2fa = true;
+    }
+
+    async disableMultisig() {
+        const contractBytes = new Uint8Array(await (await fetch('/main.wasm')).arrayBuffer());
+        const stateCleanupContractBytes = new Uint8Array(await (await fetch('/state_cleanup.wasm')).arrayBuffer());
+        let result;
+        try {
+            result = await this.disable(contractBytes, stateCleanupContractBytes);
+        } catch (e) {
+            if (e.message.includes('too large to be viewed')) {
+                throw new Error('You must wait 15 minutes between each attempt to disable 2fa. Please try again in 15 minutes.');
+            }
+            if (e.message.includes('Request was cancelled.')) {
+                throw new Error('Request was cancelled. You must wait 15 minutes to attempt disabling 2fa again.');
+            }
+            throw new Error(e.message);
+        }
+        await this.refreshAccount();
+        this.has2fa = false;
+        return result;
+    }
+
+    static async disableMultisigWithFAK({ accountId, seedPhrase, cleanupState }) {
+        const keyStore = new InMemoryKeyStore();
+        await keyStore.setKey(NETWORK_ID, accountId, KeyPair.fromString(parseSeedPhrase(seedPhrase).secretKey));
+        const connection = Connection.fromConfig({
+            networkId: NETWORK_ID,
+            provider: {
+                type: 'JsonRpcProvider',
+                args: { url: NODE_URL },
+            },
+            signer: { type: 'InMemorySigner', keyStore },
+        });
+
+        const emptyContractBytes = new Uint8Array(await (await fetch('/main.wasm')).arrayBuffer());
+        let cleanupContractBytes;
+        if (cleanupState) {
+            cleanupContractBytes = new Uint8Array(await (await fetch('/state_cleanup.wasm')).arrayBuffer());
+        }
+
+        const account = new Account2FA(connection, accountId, {
+            helperUrl: ACCOUNT_HELPER_URL,
+        });
+
+        return await account.disableWithFAK({ contractBytes: emptyContractBytes, cleanupContractBytes });
+    }
+
+    /**
+     * Returns the list of multisig function call access keys
+     */
+    async get2faLimitedAccessKeys() {
```

Comment:
This logic is from the `Account2FA` class in `near-api-js`, duplicated here to facilitate this short-term functionality.